cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaConnectDockerfile.java

```diff
@@ -61,6 +72,42 @@ public KafkaConnectDockerfile(String fromImage, Build connectBuild) {
         writer.close();
     }
 
+    /**
+     * Generates initial stage for multi-stage build
+     * @param writer        Writer for printing the Docker commands
+     * @param plugins       List of plugins which should be added to the container image
+     */
+    private void connectorPluginsPreStage(PrintWriter writer, List<Plugin> plugins) {
+        plugins.forEach(plugin -> plugin.getArtifacts().stream().filter(artifact -> artifact instanceof MavenArtifact)
+            .forEach(mvn -> {
+                checkForShellMetachars(((MavenArtifact) mvn).getGroup());
+                checkForShellMetachars(((MavenArtifact) mvn).getArtifact());
+                checkForShellMetachars(((MavenArtifact) mvn).getVersion());
+                String repo = ((MavenArtifact) mvn).getRepository() == null ? MavenArtifact.DEFAULT_REPOSITORY : maybePatchRepository(((MavenArtifact) mvn).getRepository());
+                String artifactHash = Util.sha1Prefix(((MavenArtifact) mvn).getGroup() + ((MavenArtifact) mvn).getArtifact() + ((MavenArtifact) mvn).getVersion());
+                String downloadPomCmd = String.format("curl -L --output pom.xml %s%s/%s/%s/%s-%s.pom",
+                        repo,
+                        ((MavenArtifact) mvn).getGroup().replace(".", "/"), //org.apache.camel is translated as org/apache/camel in the URL
+                        ((MavenArtifact) mvn).getArtifact().replace(".", "/"),
+                        ((MavenArtifact) mvn).getVersion(),
+                        ((MavenArtifact) mvn).getArtifact(),
+                        ((MavenArtifact) mvn).getVersion());
+
+                writer.println("FROM " + mavenImage + " AS download" + artifactHash);
+                writer.println("RUN " + downloadPomCmd + " \\");
+                writer.println("      && mvn dependency:copy-dependencies -DoutputDirectory=/tmp/artifacts/");
+                writer.println();
```

Comment:
Can we try to do this as multiple commands in a single stage? I.e. instead of stage per artifact use one stage and output directory per artifact?